### PR TITLE
[LWG 8] P2609R3 Relaxing Ranges Just A Smidge

### LINKED ISSUE #6098: [2023-02 LWG Motion 8] P2609R3 Relaxing Ranges Just A Smidge

### PR DESCRIPTION
**Pull request by burblebee**
Fixes NB US 39-099 (C++23 CD).

Fixes #6098.
Fixes cplusplus/papers#1271
Fixes cplusplus/nbballot#516

### CODE REVIEW

File: source/iterators.tex [hunk 1]
```diff
@@ -2408,6 +2411,20 @@
 There are several concepts that group requirements of algorithms that
 take callable objects\iref{func.def} as arguments.
 
+\rSec3[indirectcallable.traits]{Indirect callable traits}
+
+\pnum
+To implement algorithms taking projections,
+it is necessary to determine the projected type of an iterators value type.
```
Comment (JohelEGP) [inline source/iterators.tex:2418]:
```suggestion
it is necessary to determine the projected type of an iterator's value type.
```
Not in the paper.

File: source/iterators.tex [hunk 2]
```diff
@@ -2408,6 +2411,20 @@
...
+The exposition-only alias template \tcode{indirect-value-t<T>} denotes
```
Comment (JohelEGP) [inline source/iterators.tex:2419]: `indirect-value-t<T>` is not an alias template, it's an alias template specialization.
```suggestion
The exposition-only alias template specialization \tcode{indirect-value-t<T>} denotes
```

File: source/iterators.tex [hunk 3]
```diff
@@ -2420,48 +2437,48 @@
...
 
   template<class F, class I1, class I2 = I1>
     concept @\deflibconcept{indirect_equivalence_relation}@ =
       @\libconcept{indirectly_readable}@<I1> && @\libconcept{indirectly_readable}@<I2> &&
       @\libconcept{copy_constructible}@<F> &&
-      @\libconcept{equivalence_relation}@<F&, iter_value_t<I1>&, iter_value_t<I2>&> &&
-      @\libconcept{equivalence_relation}@<F&, iter_value_t<I1>&, iter_reference_t<I2>> &&
+      @\libconcept{equivalence_relation}@<F&, @\exposid{indirect-value-t}@<I1>, @\exposid{indirect-value-t}@<I2>> &&
+      @\libconcept{equivalence_relation}@<F&, @\exposid{indirect-value-t}@<I1>, iter_reference_t<I2>> &&
       @\libconcept{equivalence_relation}@<F&, iter_reference_t<I1>, iter_value_t<I2>&> &&
```
Comment (JohelEGP) [inline source/iterators.tex:2482]:
```suggestion
      @\libconcept{equivalence_relation}@<F&, iter_reference_t<I1>, @\exposid{indirect-value-t}@<I2>> &&
```

**Review by JohelEGP — CHANGES_REQUESTED** [review]

File: source/iterators.tex (hunk 2 above)
Comment (CaseyCarter) [reply source/iterators.tex:2419]: Maybe "For the exposition-only alias template \tcode{\exposid{indirect-value-t}}, \tcode{\exposid{indirect-value-t}<T>} denotes"? We want to say that the template is exposition-only, not that the specialization for `T` is.

Comment (JohelEGP) [reply source/iterators.tex:2419]: That's much better. As suggested:
```suggestion
For the exposition-only alias template \exposid{indirect-value-t}, \tcode{\exposid{indirect-value-t}<T>} denotes
```

File: source/iterators.tex (hunk 1 above)
Comment (burblebee) [reply source/iterators.tex:2418]: thanks! fixed

File: source/iterators.tex (hunk 2 above)
Comment (burblebee) [reply source/iterators.tex:2419]: Much better! added as a separate commit

**Review by JohelEGP — APPROVED** [review]

**Pull request merged**